rust_src/src/process.rs

```diff
@@ -134,7 +134,7 @@ pub fn get_process_lisp(name: LispObject) -> LispObject {
     if name.is_process() {
         name
     } else {
-        name.as_string_or_error();
```

Comment:
This is another spot where the intent is to verify the type.

Reply:
Yeah, that's an indication that the function signature needs to be tightened up. I added a comment about this in https://github.com/Wilfred/remacs/issues/1209#issuecomment-452903395

Reply:
The C code has things like `CHECK_FIXNUM`. Perhaps something like 
that is helpful here? `CHECK_STRING`.

I like the cast operators for simplifying the code. However, we will need to be sure they are efficient at some point.

Reply:
For now I'm trusting in the Rust propaganda that says the compiler is able to do lots of fancy type deductions while still producing fast code. We'll have to profile to know for sure. Do we have any profiling machinery set up?

Reply:
I think those `CHECK_TYPE` operators are only needed because the C code can't get any finer-grained type-wise than `Lisp_Object`. In this particular case the type should be `ProcessOrString` or whatever.

Reply:
I mostly trust the Rust compiler. It is the marshalling and demarshalling in and out of C that concerns me. It may not be a valid concern, but I know remacs feels slower in places.

Reply:
As for profiling and the list since we are C hosted and Rust is only a library I am not sure where to start. I have not had the time to research it.